test/Governance.test.ts

```diff
@@ -5,613 +5,818 @@ import {expect} from 'chai'
+            [DEVELOPER_ROLE],
+            [this.roleVotingPower]
+        )
+        await this.votesOracle.setRolesVotingPower(
+            this.delegatee2.address,
+            [LEGAL_ROLE],
+            [this.roleVotingPower]
+        )
+        await this.votesOracle.setRolesVotingPower(
+            this.delegatee3.address,
+            [TREASURY_ROLE],
+            [this.roleVotingPower]
+        )
 
-  it('Run LEGAL proposal', async function () {
-    const proposalSignatures: string[] = [
-      'addRoleMember(bytes32,address,address)'
-    ]
-    const proposalValues: string[] = ['0']
-    const proposalRoles: string[] = [LEGAL_ROLE]
-    const proposalTargets: string[] = [this.governance.address]
-    const description = 'Give voter LEGAL role'
-    const web3 = new Web3()
-    const proposalCalldatas: string[] = [
-      web3.eth.abi.encodeFunctionCall(
-        {
-          name: 'addRoleMember',
-          type: 'function',
-          inputs: [
-            {
-              type: 'bytes32',
-              name: 'role'
-            },
-            {
-              type: 'address',
-              name: 'member'
-            },
-            {
-              type: 'address',
-              name: 'proposer'
-            }
-          ]
-        },
-        [LEGAL_ROLE, this.voter.address, this.delegatee2.address]
-      )
-    ]
-    await this.runProposal(
-      this.delegatee2,
-      proposalTargets,
-      proposalValues,
-      proposalRoles,
-      proposalSignatures,
-      proposalCalldatas,
-      description
-    )
-  })
+        await this.timelock.grantRole(PROPOSER_ROLE, this.governance.address)
+        await this.timelock.grantRole(EXECUTOR_ROLE, this.governance.address)
+
+        this.runProposal = async function (
+            proposer: SignerWithAddress,
+            proposalTargets: string[],
+            proposalValues: string[],
+            proposalRoles: string[],
+            proposalSignatures: string[],
+            proposalCalldatas: string[],
+            description: string
+        ): Promise<string> {
+            await this.governance
+                .connect(proposer)
+                .propose(
+                    proposalTargets,
+                    proposalValues,
+                    proposalRoles,
+                    proposalSignatures,
+                    proposalCalldatas,
+                    description
+                )
+            const descriptionHash = ethers.utils.keccak256(
+                ethers.utils.toUtf8Bytes(description)
+            )
+            const proposalId = await this.governance.hashProposal(
+                proposalTargets,
+                proposalValues,
+                proposalCalldatas,
+                descriptionHash
+            )
+            let state = await this.governance.state(proposalId)
+            // state is Governance.ProposalState enum, state must be at Pending(0) stage here.
+            expect(state).to.equal(0)
```

Comment:
Ah.... gotcha.

suggestion: instead of a comment, an enum or constant could be used for `0`.

Rather than using comments, we can achieve full documentation by using the code, if we add the context. 
e.g. As we're in a test for governance:
```
enum ProposalState {
  pending = 0
}
```

Then the compare being `expect(state).to.equal(ProposalState.pending)`, which provides the context for the magic number, saving having to write (and keep updated) a comment in this case 🙂